src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -494,14 +736,34 @@ def forward(
 
         if inputs_embeds is None:
             # 1. Extract the input embeddings
-            inputs_embeds = self.get_input_embeddings()(input_ids)
+            # In case image_token_index is not in the embeddings (extra token but embedding don't have it)
+            for_inputs_embeds_ids = input_ids.clone()
+            for_inputs_embeds_ids[(input_ids == self.config.image_token_index)] = 0
+            inputs_embeds = self.get_input_embeddings()(for_inputs_embeds_ids)
 
             # 2. Merge text and images
-            if pixel_values is not None and input_ids.shape[1] != 1:
-                batch_size, num_patches, num_channels, height, width = pixel_values.shape
-                reshaped_pixel_values = pixel_values.view(batch_size * num_patches, num_channels, height, width)
-                image_features = self.vision_tower(reshaped_pixel_values, output_hidden_states=True)
-
+            if pixel_values is not None and input_ids.shape[1] != 1 and pixel_values.size(0) > 0:
+                # ! infer image_num_patches from image_sizes
+                image_num_patches = [
+                    image_size_to_num_patches(
+                        image_size=imsize,
+                        grid_pinpoints=self.config.image_grid_pinpoints,
+                        patch_size=self.config.vision_config.image_size,
+                    )
+                    for imsize in image_sizes
+                ]
+                # figure out if pixel_values is concatenate or stacking
+                if pixel_values.dim() == 5:
+                    # stacking: e.g [2, 5, 3, 336, 336], convert to concat
+                    _pixel_values_list = [
+                        pix_val[:num_patch] for pix_val, num_patch in zip(pixel_values, image_num_patches)
+                    ]
+                    pixel_values = torch.cat(_pixel_values_list, dim=0)
```

Comment:
Is the padding behaviour between the images and tokenizer consistent? i.e. do we correctly pad if we do padding left for tokenizers?

Reply:
patch padding is independent from tokenizer padding, and this unpadding step is only need to be consistent with what we implement padding in the image_processor, no need to split into left or right (or precisely up or down) padding.